nominal/thirdparty/pandas/_pandas.py

```diff
@@ -215,36 +221,89 @@ def _to_pandas_timestamp(timestamp: Timestamp) -> pd.Timestamp:
     return pd.Timestamp(timestamp.seconds, unit="s", tz="UTC") + pd.Timedelta(timestamp.nanos, unit="ns")
 
 
-def _to_pandas_unit(unit: ts._LiteralTimeUnit) -> str:
-    return {
-        "nanoseconds": "ns",
-        "microseconds": "us",
-        "milliseconds": "ms",
-        "seconds": "s",
-        "minutes": "m",
-        "hours": "h",
-    }[unit]
+def datasource_to_dataframe_batches(
+    datasource: DataSource,
+    channel_exact_match: Sequence[str] = (),
+    channel_fuzzy_search_text: str = "",
+    start: str | datetime | ts.IntegralNanosecondsUTC | None = None,
+    end: str | datetime | ts.IntegralNanosecondsUTC | None = None,
+    tags: dict[str, str] | None = None,
+    *,
+    channels: Sequence[Channel] | None = None,
+    num_workers: int = 1,
+    channel_batch_size: int = 20,
+    relative_to: datetime | ts.IntegralNanosecondsUTC | None = None,
+    relative_resolution: ts._LiteralTimeUnit = "nanoseconds",
+) -> Generator[pd.DataFrame, None, None]:
+    """Download a dataset to a pandas dataframe, optionally filtering for only specific channels of the dataset.
 
+    Args:
+    ----
+        datasource: The datasource to download data from
+        channel_exact_match: Filter the returned channels to those whose names match all provided strings
+            (case insensitive).
+            For example, a channel named 'engine_turbine_rpm' would match against ['engine', 'turbine', 'rpm'],
+            whereas a channel named 'engine_turbine_flowrate' would not!
+        channel_fuzzy_search_text: Filters the returned channels to those whose names fuzzily match the provided
+            string.
+        channels: List of channels to fetch data for. If provided, supercedes search parameters of
+            `channel_exact_match` and `channel_fuzzy_search_text`.
```

Comment:
Bah I don't love this function signature (I know it was there before). Generally I try to avoid function signatures where the signature itself can't enforce the calling convention. I wonder if we can use `typing.overload` to enforce that if you call with `channels` you can't use the other args?